mlir/lib/Conversion/LLVMCommon/VectorPattern.cpp

```diff
@@ -103,6 +103,11 @@ LogicalResult LLVM::detail::handleMultidimensionalVectors(
   return success();
 }
 
+static bool isVectorCompatibleType(Type type) {
```

Comment:
Can just `LLVM::isCompatibleType` be used here? It already checks for `LLVMArrayType`, `VectorType`, etc. Alternatively, there is also `LLVM::isCompatibleVectorType`, which may be useful here.

Reply:
It was using `LLVM::isCompatibleType` before, but it's too broad, I specifically want to limit this transform to scalar and vector types.

Reply:
Instead of fiddling, can you just set `PatternBenefit` on the bitcast pattern?

Reply:
While `PatternBenefit` will probably work in this specific case, I still think there is a potential problem in `vectorOneToOneRewrite` as it can generate llvm bitcasts for unsupported types like structs.